starter_code/src/movies.js

```diff
@@ -2,14 +2,132 @@
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
+/*const orderByYear = movies.sort(function(a, b) {
+  if (a.year === b.year) {
+    return a.title.localeCompare(b.title);
+  }
+  return a.year - b.year;
+});*/
+function orderByYear(allMovies) {
+  const orderByYear = allMovies.sort(function(a, b) {
+    if (a.year === b.year) {
+      return a.title.localeCompare(b.title);
+    }
+    return a.year - b.year;
+  });
+  if (allMovies.length < 1) {
+    return [];
+  }
+  return orderByYear;
+}
+
+console.log(orderByYear(movies));
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(movie) {
+  const howManyMovies = movie
+    .filter(function(movie) {
+      if (
+        movie.director === "Steven Spielberg" &&
+        movie.genre.includes("Drama")
+      ) {
+        return true;
+      }
+    })
+    .map(function(movie) {
+      return movie.title;
+    });
+  return howManyMovies.length;
+}
+
+console.log(howManyMovies(movies));
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(allMovies) {
+  const orderAlphabetically = allMovies
+    .slice()
+    .sort(function(a, b) {
+      return a.title.localeCompare(b.title);
+    })
+    .map(function(movie) {
+      return movie.title;
+    });
+
+  if (allMovies.length < 1) {
+    return [];
+  }
+  if (orderAlphabetically.length > 20) {
```

Comment:
This if statement here (if(orderAlphabetically.length > 20)) is not necessary as I'm pretty sure slice will only slice an array up to 20 (slice(0, 20)) and if an array is shorter, it will slice it accordingly. A bit of a clunky explanation but.. If you remove the if statement the code still works perfectly fine with Jasmine:)